src/slide-up-down.js

```diff
@@ -30,52 +28,33 @@ export default {
     )
   },
 
-  mounted () {
-    this.layout()
-
-    window.addEventListener('resize', this.layout)
-
-    setTimeout(() => {
-      this.isMounted = true
-    }, 0)
+  async mounted () {
+    window.addEventListener('resize', this.getHeight) // recalc height on resize window
+    await this.$nextTick()
+    this.getHeight()
   },
 
   destroyed () {
-    window.removeEventListener('resize', this.layout)
-  },
-
-  watch: {
-    active () {
-      this.layout()
-    }
+    window.removeEventListener('resize', this.getHeight)
   },
 
   computed: {
     style () {
+      const heightSize = this.active ? this.scrollHeight : 0
+
       return {
         overflow: 'hidden',
-        'transition-property': 'height',
-        height: this.isMounted ? this.maxHeight + 'px' : 'auto',
-        'transition-duration': this.duration + 'ms'
+        transitionProperty: 'height',
```

Comment:
These changes are just changes in coding style and are not relevant to the content of the PR.